pkg/kvcache/kvblock/token_processor.go

```diff
@@ -84,27 +73,60 @@ func NewChunkedTokenDatabase(config *TokenProcessorConfig) TokenProcessor {
 	}
 }
 
-// getInitHash returns the initial hash.
-func (db *ChunkedTokenDatabase) getInitHash() string {
-	return ""
+// getInitHash returns the root parent hash.
+func (db *ChunkedTokenDatabase) getInitHash() *uint64 {
+	if db.initHash != nil {
+		return db.initHash
+	}
+
+	encMode, err := cbor.CanonicalEncOptions().EncMode() // deterministic
+	if err != nil {
+		klog.FromContext(context.Background()).Error(err, "failed to create CBOR encoder")
+		return nil
+	}
+
+	b, err := encMode.Marshal(db.HashSeed)
+	if err != nil {
+		klog.FromContext(context.Background()).Error(err, "failed to marshal payload to CBOR")
+		return nil
+	}
+
+	sum := sha256.Sum256(b)
+	hashVal := binary.BigEndian.Uint64(sum[24:])
+	db.initHash = &hashVal
+	return db.initHash
 }
 
-// hash computes the SHA-256 hash of the concatenation of the prefixHash and the binary
-// representation of the tokens slice. It returns the hex-encoded string.
-func (db *ChunkedTokenDatabase) hash(tokens []uint32, prefixHash string) string {
-	buf := new(bytes.Buffer)
-	// write the prefixHash bytes (ASCII encoding)
-	buf.WriteString(prefixHash)
-	// write each token to the buffer as binary data (using 64-bit big-endian format)
-	for _, token := range tokens {
-		// convert token to int64 for binary consistency
-		// LittleEndian is important to match the Python code
-		if err := binary.Write(buf, binary.LittleEndian, token); err != nil {
-			klog.FromContext(context.Background()).Error(err, "failed to write token to buffer")
-		}
+// hash computes a uint64 hash (lower 64 bits of SHA256).
+// The format, serialization and hashing is aligned with that of vLLM.
+func (db *ChunkedTokenDatabase) hash(parent uint64, tokens []uint32, extra interface{}) uint64 {
```

Comment:
Q: is there really nothing that can be said about `extra`? 
Is it a byte array, does it implement Stringer, ....?
Seems to at least be cbor.Marshal enable?

Reply:
Delegating this to follow-up work as originally intended. In general this field will contain information such as LoRA model-id and multi-modal data. Opening issue.